### ISSUE TITLE
Auto p corrupting paragraph blocks.

### ISSUE THREAD

**Issue by nbcsteveb** [WordPress/gutenberg#64353]
### Description

If I save an article such as:
````
wp_insert_post(['post_content'=>'<!-- wp:paragraph --><p dir="ltr">Hello, world!</p><! -- /wp:paragraph -->']);
````

When you open the article and save as draft, [this function](https://github.com/WordPress/WordPress/blob/25cd14c5092d3e9fbc6df5781e18f058c347c67e/wp-includes/js/dist/blocks.js#L13722) (or something around it) is updating this content to:

````
<!-- wp:paragraph -->
<p><p dir="ltr">Hello, world!</p></p>
<!-- /wp:paragraph -->
````

Which Gutenberg then tells the user is an invalid block of content:

<img width="899" alt="image" src="https://github.com/user-attachments/assets/2eb19422-e29c-4cb0-b4b2-d9bec8291f8e">

And when resolving:
<img width="496" alt="image" src="https://github.com/user-attachments/assets/47e9eb3d-e2ac-49c1-87cb-db5014e2db42">

### Step-by-step reproduction instructions

1. Save article: ````wp_insert_post(['post_content'=>'<!-- wp:paragraph --><p dir="ltr">Hello, world!</p><! -- /wp:paragraph -->']);````
2. Open article in WP and save as draft.
3. Refresh the page.
4. Notice the warning about invalid content.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by talldan** [comment]
I wasn't able to reproduce the exact same issue. For me, the block is show as invalid straight away (step 2) and I don't see extra paragraphs. It might be a difference in configuration. I don't think the paragraph knows how to handle the `dir` attribute, so that's why I get a validation issue.

If I use markup that I created in the editor using the 'Language' format I'm not seeing any issues:
```php
wp_insert_post( array( 'post_content' => '<!-- wp:paragraph --><p><bdo lang="" dir="ltr">Hello, world!</bdo></p><!-- /wp:paragraph -->' ) );
```

**Comment by miminari** [comment]
I also tested on WordPress 6.6.1, and it wasn't reproduced. 
Likewise, I could not proceed to Step 2.

At the same time, I tried the suggested code adding `lang` attribute and it worked fine.
> ```
> wp_insert_post( array( 'post_content' => '<p><bdo lang="" dir="ltr">Hello, world!</bdo></p>' ) );
> ```

![SS_2024-08-14 9 51 38](https://github.com/user-attachments/assets/3736712a-ffa6-4378-9e61-1716babcd209)

Some assumptions might be different, but I wasn't able to reproduce too.
